src/test/java/com/gepardec/mega/zep/rest/service/AttendanceServiceTest.java

```diff
@@ -0,0 +1,157 @@
+                .id(1)
+                .date(LocalDate.of(2018,12,11))
+                .from(LocalTime.of(8,0,0))
+                .to(LocalTime.of(9,15,0))
+                .employeeId("001-duser")
+                .projectId(1)
+                .projectTaskId(1)
+                .duration(1.25)
+                .billable(true)
+                .workLocation(null)
+                .workLocationIsProjectRelevant(false)
+                .activity("besprechen")
+                .vehicle(null)
+                .directionOfTravel(null)
+                .build(),
+            ZepAttendance.builder()
+                .id(2)
+                .date(LocalDate.of(2015,2,11))
+                .from(LocalTime.of(9,0,0))
+                .to(LocalTime.of(13,45,0))
+                .employeeId("001-duser")
+                .projectId(1)
+                .projectTaskId(2)
+                .duration(4.75)
+                .billable(true)
+                .workLocation(null)
+                .workLocationIsProjectRelevant(true)
+                .activity("bearbeiten")
+                .vehicle(null)
+                .directionOfTravel(null)
+                .build()
+        );
+
+        List<ZepAttendance> attendances = attendanceService.getBillableAttendancesForUserAndMonth("001-duser", LocalDate.of(2018,12,11));
+        assertThat(List.of(attendances.get(0), attendances.get(1))).usingRecursiveComparison().isEqualTo(attendancesReference);
+    }
+
+    @Test
+    public void getAttendancesPaginated() {
+        List<ZepAttendance> attendances = attendanceService.getBillableAttendancesForUserAndMonth("001-duser", LocalDate.now());
+        IntStream.range(0, 3).forEach(
+                i -> assertThat(attendances.get(i).id()).isEqualTo(i + 1)
+        );
+    }
+    @Test
+    public void notBillableAttendances_thenReturnNull() {
+        List<ZepAttendance> attendances = attendanceService.getBillableAttendancesForUserAndMonth("001-duser", LocalDate.now());
+        assertThat(attendances.size()).isEqualTo(3);
+    }
+
+    @Test
+    @Disabled
+    public void extractCorrectMonthFromGivenDate_thenCallPaginator() {
+        ArgumentCaptor<String> startDateCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> endDateCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> usernameCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Function> functionCaptor = ArgumentCaptor.forClass(Function.class);
+
+        ResponseParser mockedResponseParser = Mockito.mock(ResponseParser.class);
+
+        when(mockedResponseParser.retrieveAll(any(), any()))
+                .thenReturn(new ArrayList<>());
+
+        //Call the Method under test
+        attendanceService.getAttendanceForUserAndMonth("username", LocalDate.of(2021, 1, 10));
+
+        //Retrieve the function called in the method under test
+        verify(mockedResponseParser.retrieveAll(functionCaptor.capture(), any()));
```

Comment:
@sgartner03 complete verify

Reply:
is disabled because not working properly